src/hotspot/share/opto/vectornode.cpp

```diff
@@ -1042,10 +1042,67 @@ Node* VectorNode::try_to_gen_masked_vector(PhaseGVN* gvn, Node* node, const Type
   }
 }
 
+bool VectorNode::can_swap_inputs() {
+  // Must be a binary operation.
+  if (req() != 3) {
+    return false;
+  }
+
+  // Predicated vector operations are sensitive to ordering of inputs.
+  // When the mask corresponding to a vector lane is false then
+  // the result of the operation is corresponding lane of its first operand.
+  //   i.e. RES = VEC1.lanewise(OPER, VEC2, MASK) is semantically equivalent to
+  //        RES = VEC1.BLEND(VEC1.lanewise(OPER, VEC2), MASK)
+  // Here, first operand "VEC1" represents both source and destination vector.
+  if (is_predicated_vector()) {
+    return false;
+  }
+
+  switch(Opcode()) {
+    case Op_AddVB:
+    case Op_AddVS:
+    case Op_AddVI:
+    case Op_AddVL:
+    case Op_AddVF:
+    case Op_AddVD:
+
+    case Op_MulVB:
+    case Op_MulVS:
+    case Op_MulVI:
+    case Op_MulVL:
+    case Op_MulVF:
+    case Op_MulVD:
+
+    case Op_MaxV:
+    case Op_MinV:
+    case Op_XorV:
+    case Op_OrV:
+    case Op_AndV:
+    case Op_UMinV:
+    case Op_UMaxV:
+
+    case Op_AndVMask:
+    case Op_OrVMask:
+    case Op_XorVMask:
+
+    case Op_SaturatingAddV:
+      if (in(1)->_idx > in(2)->_idx) {
+        return true;
+      }
```

Comment:
@jatin-bhateja can you explain this / add a comment to the code?

```suggestion
      if (in(1)->_idx > in(2)->_idx) {
        // Sort the inputs of commutative non-predicated vector operations to help value numbering.
        return true;
      }
```